security/pkg/pki/ra/common.go

```diff
@@ -0,0 +1,100 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package ra
+
+import (
+	"fmt"
+	"time"
+
+	certificatesv1beta1 "k8s.io/client-go/kubernetes/typed/certificates/v1beta1"
+
+	"istio.io/istio/security/pkg/pki/util"
+	caserver "istio.io/istio/security/pkg/server/ca"
+)
+
+// RegistrationAuthority : Registration Authority interface.
+type RegistrationAuthority interface {
+	caserver.CertificateAuthority
+}
+
+// CaExternalType : Type of External CA integration
+type CaExternalType string
+
+// IstioRAOptions : Configuration Options for the IstioRA
+type IstioRAOptions struct {
+	// ExternalCAType: Integration API type with external CA
+	ExternalCAType CaExternalType
+	// DefaultCertTTL: Default Certificate TTL
+	DefaultCertTTL time.Duration
+	// MaxCertTTL: Maximum Certificate TTL that can be requested
+	MaxCertTTL time.Duration
+	// CaCertFile : File containing PEM encoded CA root certificate of external CA
+	CaCertFile string
+	// CaSigner : To indicate custom CA Signer name when using external K8s CA
+	CaSigner string
+	// VerifyAppendCA : Whether to use caCertFile containing CA root cert to verify and append to signed cert-chain
+	VerifyAppendCA bool
```

Comment:
I think verifying is one thing, and appending is another. Do we want to control them in one flag?

Reply:
Agree. Will need to modify this in a future PR. The flag was tailored to what we do today for Istio CA and Kubernetes External CA